build.gradle

```diff
@@ -0,0 +1,405 @@
+    ext.ideaSdkCodeline = codeLine
+    ext.ideaSdkBuildNumber = buildNumber
+    ext.ideaSdkPatchNumber = patchNumber
+
+    // Lookup table for properties files.  Add new versions here.
+    def propertiesFile = ""
+    switch (Integer.valueOf(codeLine, 10)) {
+        case 183: propertiesFile = "idea_v18.properties"; break
+        case 182: propertiesFile = "idea_v18.properties"; break
+        case 181: propertiesFile = "idea_v18.properties"; break
+        case 173: propertiesFile = "idea_v17.3.properties"; break
+        case 172: propertiesFile = "idea_v17.properties"; break
+        case 171: propertiesFile = "idea_v17.properties"; break
+        case 163: propertiesFile = "idea_v16.properties"; break
+        case 162: propertiesFile = "idea_v16.properties"; break
+        case 161: propertiesFile = "idea_v16.properties"; break
+        case 145: propertiesFile = "idea_v16.properties"; break
+    }
+    def properties = new Properties()
+    if (0 != propertiesFile.size()) {
+        file("${rootDir}/properties/" + propertiesFile).withInputStream { properties.load(it) }
+    } else {
+        println "WARNING!: No Properties file found"
+    }
+    properties.setProperty('idea.sdk.version', "$ideaVersion")
+    properties.setProperty('idea.sdk.type', buildType)
+    properties.setProperty('idea.sdk.codeline', codeLine)
+    properties.setProperty('idea.sdk.build.number', buildNumber)
+    properties.setProperty('idea.sdk.patch.number', patchNumber)
+    properties.setProperty('replace.with.plugin.autogen.warning', "${pluginAutogenWarning}")
+    properties.setProperty('plugin.dev.version', project.hasProperty('devBuild') ? "${project.devBuild}" : '')
+
+    return properties
+}
+
+tasks.withType(Test) {
+    // force all tests to run every time
+    dependsOn "cleanTest"
+
+    // output summary and any failed or skipped tests
+
+    // from https://stackoverflow.com/questions/3963708/gradle-how-to-display-test-results-in-the-console-in-real-time
+    testLogging {
+        events TestLogEvent.FAILED,
+                TestLogEvent.SKIPPED
+
+        exceptionFormat TestExceptionFormat.FULL
+        showExceptions true
+        showCauses true
+        showStackTraces true
+
+        info.events = debug.events
+        info.exceptionFormat = debug.exceptionFormat
+
+        afterSuite { desc, result ->
+            if (!desc.parent) { // will match the outermost suite
+                def output = "Results: ${result.resultType} (${result.testCount} tests, ${result.successfulTestCount} successes, ${result.failedTestCount} failures, ${result.skippedTestCount} skipped)"
+                def startItem = '|  ', endItem = '  |'
+                def repeatLength = startItem.length() + output.length() + endItem.length()
+                println('\n' + ('-' * repeatLength) + '\n' + startItem + output + endItem + '\n' + ('-' * repeatLength))
+            }
+        }
+    }
+}
+
+
+task checkForMissingRunConfigurations() {
```

Comment:
When does this get called?  Or is it supposed to be a menu-only thing?

Reply:
Gradle has different phases; the logic in this task is in the configuration phase and will be executed whenever Gradle “evaluate/reads” this task. It was the easiest way to get logic executed and runConfigurations created for intelliJ without having to get the user to build or run any tasks.

Placing it in the Initialization phase could also be an option
https://docs.gradle.org/current/userguide/build_lifecycle.html